runtime/src/utils/buffer.rs

```diff
@@ -111,11 +114,12 @@ impl<B: Blob> Buffer<B> {
         let bytes_to_read = std::cmp::min(self.buffer_size as u64, blob_remaining) as usize;
 
         // Read the data - we only need a single read operation since we know exactly how much data is available
-        self.blob
-            .read_at(&mut self.buffer[..bytes_to_read], self.blob_position)
+        let buffer = std::mem::take(&mut self.buffer);
+        self.buffer = self
+            .blob
+            .read_at(BytesMut::from(&buffer[..bytes_to_read]), self.blob_position)
             .await?;
         self.buffer_valid_len = bytes_to_read;
-
         Ok(bytes_to_read)
     }
 
```

Comment:
nit: might want to change to `read_at` for consistency

Reply:
If it's alright with you I'd rather leave this for a separate PR. I made an issue. https://github.com/commonwarexyz/monorepo/issues/936